### DISCUSSION TITLE
bump Transmission

### DISCUSSION THREAD

**Discussion by AkteX** [Freetz-NG/freetz-ng#751] [Unerfüllte Wünsche] [answered]
Hallo,
nach dem Hinzufügen eines Torrents stürzt Transmission 3.0 ab. Ich habe Transmission mit --log-debug gestartet. In der Log Datei steht als letzter Eintrag "Unable to create session lock file (13): Permission denied (session-id.c:96)". Eine Fehlermeldung gibt es unter: [https://github.com/transmission/transmission/issues/1391](https://github.com/transmission/transmission/issues/1391). Sollte man den Patch einbauen oder Transmission aktualisieren? Inzwischen gibt es die Version 4.0.1.

**Comment by fda77** [comment]
Weshalb sollte die v3 aufhören zu funktionieren nur weil es nun die v4 neu gibt? Ich vermute eher dass du den daemon unter einem benutzer laufen hast der in dem Verzeichnis in das diese Lockdatei soll keine Schreibrechte hat

**Comment by AkteX** [comment]
Der Fehlerreport mit empfohlenem Patch ist für v3.0. Eine Aktualisierung auf v4.0.1 würde den Patch überflüssig machen. Das Basisverzeichnis ist für bittorrent.users schreibbar. Der verlinkte Fehlerbericht bezieht sich auf die bei einigen Kernel Versionen nicht verfügbare Lockmöglichkeit. Das beschriebene Fehlverhalten tritt bei meiner 6591 mit FW 7.50 auf.

**Comment by fda77** [comment]
Na wenn nur dieser kernel nicht will.
Pass aber beim bump auf, da scheint sich einiges mit den dependencies verändert haben, libdeflate statt zlib, keine libiconv mehr. Und irgend was mit einer libutp

**Comment by AkteX** [comment]
Das ist nichts für mich. Da muss ein Experte ran.

**Comment by fda77** [comment]
Beim Freetz gibts aktuell leider keine.
Für grössere Änderungen mit libs muss man auf jeden fall testen, und es kann auch sein dass es dann auf den älteren Geräten nicht mehr will

**Comment by AkteX** [comment]
Wäre vielleicht wäre eine genauere Analyse sinnvoll, um den Fehler bei der Version 3.0 zu fixen?
Folgender Fehler ist reproduzierbar:
1. Torrent starten
2. Laufenden Torrent anhalten

Strace Ausgabe:
```clock_gettime(CLOCK_MONOTONIC_COARSE, {tv_sec=454, tv_nsec=666038661}) = 0
gettimeofday({tv_sec=1678020326, tv_usec=914076}, NULL) = 0
gettimeofday({tv_sec=1678020326, tv_usec=914174}, NULL) = 0
epoll_wait(6,  <unfinished ...>)        = ?
+++ killed by SIGSEGV +++
```
Letzte Einträge im Transmission Debug Log:
```
[2023-03-05 14:08:59.722] Saved "/var/media/ftp/data/bittorrent/config/resume/a97a027c2bfaf2b0cc276a751cc66026a8cd7f36.resume" (variant.c:1>
[2023-03-05 14:08:59.728] <Torrent Name> Announcing to tracker (announcer.c:1723)>
[2023-03-05 14:08:59.731] DHT Attempting bootstrap from dht.transmissionbt.com (tr-dht.c:299)
```
Hast Du eine Idee?

**Comment by fda77** [comment]
Wenn man strace mit vielen "f" aufruft werden auch die Childs vetrachtet ("-ffffff")
Das strace ist bisschen kurz.
Aber das "keine Berechtigung" ist doch soweit klar. Schau wo was angelegt werden soll zb /tmp oder /run , welcher user du dem daemon gegeben hast und welche Berechtigungen es im Dateisystem  sind
ALternativ, gibt es auch server ohne DHT? Vielleicht geht das nciht weil irgend was nich eincompiliert ist.
Ging das bei dir überhaupt schon jemals?

**Comment by AkteX** [comment]
Keine Berechtigung sollte folgender Eintrag im strace sein:
```
[pid 32197] open("/tmp/tr_session_id_18gkRyw8c3GgVgBUn73bFtXDPH8NY6rv187O3UgSFbWM4H2K", O_RDWR|O_CREAT|O_LARGEFILE|O_CLOEXEC, 0600) = -1 EACCES (Permission denied)
```
Es ist mir nicht klar, warum diese Fehlermeldung kommt, da /tmp 777-Rechte gesetzt sind. Der Session lock soll scheinbar gesetzt werden, wenn ich mich mit Transmission Remote GUI mit transmission auf der Box verbinde. Dies scheint jedoch nicht die Ursache der Abstürze zu sein.

Die Abstürze treten immer auf, wenn Torrents geladen werden. Ein direkter Zusammenhang mit bestimmten Aktionen ist nicht ersichtlich. Die Abstürze beim Stoppen sind leider auch nur in bestimmten Situationen reproduzierbar. Je größer der Torrent, desto eher treten "spontane?" Abstürze auf. Das Verhalten ist sehr instabil.

**Comment by fda77** [comment]
> Ging das bei dir überhaupt schon jemals?

???

es geht nicht um die rechte von /tmp sondern von dieser datei darin.
davon abgesehen ist /tmp ein link und links haben grundsätzlich keine rechte 

> "spontane?" Abstürze

 das könnte ram oder cpu mangel sein. Das was avm da zusammenfummelt ist auch nicht sooo belastbar wir ein normales linux. syslog könnte helfen

**Comment by AkteX** [comment]
Ja, Transmission hatte mit der 7530 ohne Abstürzen funktioniert.

Längere Strace Ausgabe mit -ffffff
```
[pid 11707] ioctl(35, FIONREAD, [3157991]) = 0
[pid 11707] readv(35, [{iov_base="T\216\332\333^\205p\304\372u\343\334J\23\217\305\350\355*.h2\236\306\256\t\r2\304m\263d"..., iov_len=3000}], 1) = 3000
[pid 11707] gettimeofday({tv_sec=1678036630, tv_usec=330726}, NULL) = 0
[pid 11707] gettimeofday({tv_sec=1678036630, tv_usec=330891}, NULL) = 0
[pid 11707] ioctl(34, FIONREAD, [1818491]) = 0
[pid 11707] readv(34, [{iov_base=":,\272\370\242W+{}\371\32\305P\5\265\365n_\330>\2347\311\307\326>\353`\376\30\253\334"..., iov_len=3000}], 1) = 3000
[pid 11707] gettimeofday({tv_sec=1678036630, tv_usec=331412}, NULL) = 0
[pid 11707] gettimeofday({tv_sec=1678036630, tv_usec=331552}, NULL) = 0
[pid 11707] ioctl(27, FIONREAD, [2289978]) = 0
[pid 11707] readv(27, [{iov_base="`\34\327\316ww\367\277\37\225C\356rSb\300Rw\347\vP\v\311\357:,^{:\1q\10"..., iov_len=3000}], 1) = 3000
[pid 11707] gettimeofday({tv_sec=1678036630, tv_usec=332007}, NULL) = 0
[pid 11707] gettimeofday({tv_sec=1678036630, tv_usec=332145}, NULL) = 0
[pid 11707] ioctl(38, FIONREAD, [2515329]) = 0
[pid 11707] readv(38, [{iov_base="\3\257X\214\370\271+\230NN\326\222\31\327\347V\346\203p~\351\227\275\237\313\235kn#\261\37\27"..., iov_len=3000}], 1) = 3000
[pid 11707] gettimeofday({tv_sec=1678036630, tv_usec=332652}, NULL) = 0
[pid 11707] gettimeofday({tv_sec=1678036630, tv_usec=332802}, NULL) = 0
[pid 11707] ioctl(23, FIONREAD, [421])  = 0
[pid 11707] readv(23, [{iov_base="4\223uK\335\220\377\345\20\304D7{\203\6o\370{\203\244\240\277$D\247\260\236\"\320fl\260"..., iov_len=392}, {iov_base="Iu+jy\37}\20\322\301\10!\226h\3701\0223\302[\26\304\245\300\355\244 \303\364", iov_len=29}], 2) = 421
[pid 11707] gettimeofday({tv_sec=1678036630, tv_usec=333264}, NULL) = 0
[pid 11707] --- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x9d} ---
[pid 11705] <... epoll_wait resumed> <unfinished ...>) = ?
[pid 11766] <... nanosleep resumed>NULL) = ? ERESTART_RESTARTBLOCK (Interrupted by signal)
[pid 11769] <... _newselect resumed> <unfinished ...>) = ?
[pid 11706] <... read resumed> <unfinished ...>) = ?
[pid 11769] +++ killed by SIGSEGV +++
[pid 11766] +++ killed by SIGSEGV +++
[pid 11706] +++ killed by SIGSEGV +++
[pid 11707] +++ killed by SIGSEGV +++
+++ killed by SIGSEGV +++
```

**Comment by fda77** [comment]
Das sagt mir leider gar nichts

google sagt:
> #define [SEGV_MAPERR](https://elixir.bootlin.com/linux/v5.15.8/C/ident/SEGV_MAPERR)	1	/* address not mapped to object */

sieht nach programmierfehler aus.

was hast du denn gemacht damit es nicht mehr funktioniert?

und wie seind die rechte des verzeichnisses auf das /tmp zeigt?

und warum nimmst du statt /tmp nicht was anders?

/tmp ist ram -> /tmp voll -> ram auch -> "spontane" dinge

**Comment by AkteX** [comment]
Ich habe keine Änderungen gemacht. Ist so wie ich freetz-ng aus Github erhalte.
/tmp -> /var/tmp root.root 755, dann wird transmission dort nichts schreiben können.
Das tmp Verzeichnis wird nur zum Schreiben der Session-ID verwendet. Das Basisverzeichnis für Transmission ist /var/media/ftp/data/bittorrent und befindet sich auf einer USB Festplatte.

Hier ist ein Beispiel, wo die Verwendung unterschiedlicher Compileroptionen für Programm und Library zu einem SEGV_MAPERR geführt hat: https://community.hpe.com/t5/languages-and-scripting/sigsegv-segv-maperr-address-not-mapped-to-object/td-p/6170827

Hier ist ein Beipiel mit einer zu kleinen pthread Stacksize:
https://community.hpe.com/t5/hp-ux-general/receiving-error-quot-segv-maperr-address-not-mapped-to-object/m-p/3964313#M127889

**Comment by AkteX** [comment]
Das ist die zweite Applikation nach Midnight Commander, welche mit Segmentation Faults bei 6591 und ähnlichen Boxen abstürzt. Könnte es vielleicht ähnliche Ursachen geben?

**Comment by fda77** [comment]
Bevor du jeden Tag ein weiteres programm aufzählst: Was geht sonst auch nicht? Und wie sieht das auf deinen anderen geräten aus?
Mit fällt spontan ein:
 - pebkac
 - hardware kaputt
 - komische scripte (zb von sammy-meister)

davon abgesehen ist das "keine berechtigung" was anderes und klar wie man es zu beheben hat

**Comment by AkteX** [comment, marked as answer]
Hardware kaputt: Nein, ich habe 2 6591
Komische Scripte: Nein, verwende ich nicht

Ich scheine eine stabile Konfiguration für Transmission auf der 6591 gefunden zu haben:
1. OpenSSL Lib version 1.1.1
2. Libcurl mit SSL aber **mbed TLS** anstatt OpenSSL!

Der zweite Punkt hat die Stabilität von Transmission hergestellt. Könnte es ein Problem bei der libcurl mit OpenSSL geben?

Ein chmod 777 /var/tmp kann man sicher überall einbauen. Wo würdest Du das machen?

**Comment by fda77** [comment]
Wenn mbedtls verwendet wird, warum spiel dann eine gewisse openssl version eine rolle?

Die /tmp berechtigungen macht avm öfter kaputt, zb nach einem reconnect. libctlmgr kann das verhindern, wurde aber ab fos 7.2 nicht mehr gebaucht: https://github.com/Freetz-NG/freetz-ng/blob/master/make/libs/libctlmgr/src/libctlmgr.c#L109

**Comment by fda77** [comment]
Sag mal, du hast doch oben behauptet überhaupt nichts verändert zu haben:

> Ich habe keine Änderungen gemacht. Ist so wie ich freetz-ng aus Github erhalte.

Aber dann heisst es:

> Das beschriebene Fehlverhalten tritt bei meiner 6591 mit FW 7.50 auf.

Diese 7.5 gibt es seit 3 Wochen und ist eine GRUNDLEGENDE änderung!
Ich hab echt keine Lust mehr auf diese schwachsinnige Scheisse

**Reply by fda77** [reply]
Diese Diskussion ist gelockt weils mir zu dumm ist
